src/url-loader/url-resolver.ts

```diff
@@ -12,13 +12,16 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {posix} from 'path';
-import {resolve as urlLibResolver} from 'url';
+import * as path from 'path';
+import {format as urlLibFormat, resolve as urlLibResolver} from 'url';
 
 import {parseUrl} from '../core/utils';
 import {PackageRelativeUrl, ScannedImport} from '../index';
 import {FileRelativeUrl, ResolvedUrl} from '../model/url';
 
+const sharedRelativeUrlProperties =
```

Comment:
I think it's safe to drop `port` and `hostname` here, as `host` is a combination of the two.

Reply:
Also document this please :)

Reply:
Okay dropped port and hostname in favor of just host.  This brings the total number of properties I consider to four.  As such I decided against an array of them and just explicitly reference where used.  It makes the code clearer and I also have better type checking for the property values whereas previously only checked for null instead of null|undefined-- now I can just check non-emptiness by testing whether values are of the specific types for the properties.